htdocs/core/class/notify.class.php

```diff
@@ -314,218 +317,346 @@ function send($notifcode, $object)
-							$filepdf = '';
-						}
-						else
-						{
-							$filepdf = $pdf_path;
-						}
-
-						$message = $outputlangs->transnoentities("YouReceiveMailBecauseOfNotification",$application,$mysoc->name)."\n";
-						$message.= $outputlangs->transnoentities("YouReceiveMailBecauseOfNotification2",$application,$mysoc->name)."\n";
-						$message.= "\n";
-						$message.= $mesg;
-						if ($link) $message=dol_concatdesc($message,$urlwithroot.$link);
-
-						$parameters=array('notifcode'=>$notifcode, 'sendto'=>$sendto, 'replyto'=>$replyto, 'file'=>$file, 'mimefile'=>$mimefile, 'filename'=>$filename);
-						$reshook=$hookmanager->executeHooks('formatNotificationMessage',$parameters,$object,$action);    // Note that $action and $object may have been modified by some hooks
-						if (empty($reshook))
-						{
-							if (! empty($hookmanager->resArray['subject'])) $subject.=$hookmanager->resArray['subject'];
-							if (! empty($hookmanager->resArray['message'])) $message.=$hookmanager->resArray['message'];
-						}
-
-						$mailfile = new CMailFile(
-							$subject,
-							$sendto,
-							$replyto,
-							$message,
-							array($file),
-							array($mimefile),
-							array($filename[count($filename)-1]),
-							'',
-							'',
-							0,
-							-1
-						);
-
-						if ($mailfile->sendfile())
-						{
-							if ($obj->type_target == 'touserid') {
-	 							$sql = "INSERT INTO ".MAIN_DB_PREFIX."notify (daten, fk_action, fk_soc, fk_user, type, objet_type, type_target, objet_id, email)";
-								$sql.= " VALUES ('".$this->db->idate(dol_now())."', ".$notifcodedefid.", ".($object->socid?$object->socid:'null').", ".$obj->cid.", '".$obj->type."', '".$object_type."', '".$obj->type_target."', ".$object->id.", '".$this->db->escape($obj->email)."')";
-
-							}
-							else {
-								$sql = "INSERT INTO ".MAIN_DB_PREFIX."notify (daten, fk_action, fk_soc, fk_contact, type, objet_type, type_target, objet_id, email)";
-								$sql.= " VALUES ('".$this->db->idate(dol_now())."', ".$notifcodedefid.", ".($object->socid?$object->socid:'null').", ".$obj->cid.", '".$obj->type."', '".$object_type."', '".$obj->type_target."', ".$object->id.", '".$this->db->escape($obj->email)."')";
-
-							}
-							if (! $this->db->query($sql))
-							{
-								dol_print_error($this->db);
-							}
-						}
-						else
-						{
-							$error++;
-							$this->errors[]=$mailfile->error;
-						}
-					}
-					else
-				  {
-						dol_syslog("No notification sent for ".$sendto." because email is empty");
-					}
-					$i++;
-				}
-			}
-			else
-			{
-				dol_syslog("No notification to thirdparty sent, nothing into notification setup for the thirdparty socid = ".$object->socid);
-			}
-		}
-		else
-		{
-	   		$error++;
-			$this->errors[]=$this->db->lasterror();
-			dol_syslog("Failed to get list of notification to send ".$this->db->lasterror(), LOG_ERR);
-	   		return -1;
-		}
+        if (! empty($object->socid)) {
```

Comment:
If you add a test here, the second part that is a union to get target to users id will not be included.
If $object->socid is not defined, the test
$sql .= " AND s.rowid = ".$object->socid;
can be replaced with 
$sql .= " AND s.rowid = ".($object->socid>0?$object->socid:0);

Reply:
change done, thanks